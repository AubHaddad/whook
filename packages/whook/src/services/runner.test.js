import initHandlerRunner from './runner';

describe('initHandlerRunner', () => {
  const log = jest.fn();
  const handler = jest.fn();

  beforeEach(() => {
    log.mockReset();
    handler.mockReset();
  });

  it('should work with resolving handler', async () => {
    const handlerRunner = await initHandlerRunner({
      handler,
      parameters: { test: 'test' },
      log,
    });
    const result = await handlerRunner();

    expect({
      result,
      logCalls: log.mock.calls.filter(args => 'error' !== args[0]),
      handlerCalls: handler.mock.calls,
    }).toMatchSnapshot();
  });

  it('should work with rejecting handler', async () => {
    handler.mockRejectedValueOnce(new Error('E_ERROR'));

    const handlerRunner = await initHandlerRunner({
      handler,
      parameters: { test: 'test' },
      log,
    });
    const result = await handlerRunner();

    expect({
      result,
      logCalls: log.mock.calls.filter(args => 'error' !== args[0]),
      handlerCalls: handler.mock.calls,
    }).toMatchSnapshot();
  });
});
