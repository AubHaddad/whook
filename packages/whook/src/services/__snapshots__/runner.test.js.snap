// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initHandlerRunner should work with rejecting handler 1`] = `
Object {
  "handlerCalls": Array [
    Array [
      Object {
        "test": "test",
      },
    ],
  ],
  "logCalls": Array [
    Array [
      "info",
      "parameters",
      Object {
        "test": "test",
      },
    ],
  ],
  "result": undefined,
}
`;

exports[`initHandlerRunner should work with resolving handler 1`] = `
Object {
  "handlerCalls": Array [
    Array [
      Object {
        "test": "test",
      },
    ],
  ],
  "logCalls": Array [
    Array [
      "info",
      "parameters",
      Object {
        "test": "test",
      },
    ],
    Array [
      "info",
      "response",
      undefined,
    ],
  ],
  "result": undefined,
}
`;
